### DISCUSSION TITLE
Auto-align with embedded elements

### DISCUSSION THREAD

**Discussion by gscht** [clientIO/joint#2002] [Q&A] [answered]
Hi everyone!

We are using version 3.4.1 of the library. We have embedded elements in our graph, meaning a `Rectangle` contains another `Rectangle`.

When calling `joint.layout.DirectedGraph.layout(graph)`, we get an error in the console:
```
Uncaught TypeError: Cannot set properties of undefined (setting 'rank')
    at dfs (util.js:51:19)
    at util.js:42:14
    at arrayMap (_arrayMap.js:16:21)
    at Object.map (map.js:50:10)
    at dfs (util.js:41:24)
    at util.js:42:14
    at arrayMap (_arrayMap.js:16:21)
    at Object.map (map.js:50:10)
    at dfs (util.js:41:24)
    at arrayEach (_arrayEach.js:15:9)
```
![Screenshot 2023-01-23 at 10 23 30](https://user-images.githubusercontent.com/4086321/214005327-939304b9-ab47-44bb-b8c3-fabb3573a9e6.png)

Full demo: [layout_issue.zip](https://github.com/clientIO/joint/files/10478312/layout_issue.zip)

Any help would be appreciated very much! 🙏🏻 Thank you!

**Comment by jamesgeorgewilliams** [comment, marked as answer]
Duplicate of https://github.com/clientIO/joint/issues/455

https://jsfiddle.net/kumilingus/yfbc5knv/
